rllib/evaluation/collectors/agent_collector.py

```diff
@@ -266,6 +269,8 @@ def add_action_reward_next_obs(self, input_values: Dict[str, TensorType]) -> Non
                 or k.startswith("state_out_")
                 or (k == SampleBatch.ACTIONS and not self.disable_action_flattening)
             ):
+                if k == SampleBatch.ACTIONS and not self.disable_action_flattening:
```

Comment:
nit: we can rewrite the code to avoid repetition of the condition in two place: 
```
should_flatten_action_key = (k == SampleBatch.ACTIONS and not self.disable_action_flattening)
if should_flatten_action_key:
     v = flatten_to_single_ndarray(v)

if x or y or should_flatten_action_key:
     self.buffers[k][0].append(v)
```